src/internal/connector/graph/service.go

```diff
@@ -319,3 +365,181 @@ func RunWithRetry(run func() error) error {
 
 	return support.ConnectorStackErrorTraceWrap(err, "maximum retries or unretryable")
 }
+
+// RetryHandler handles transient HTTP responses and retries the request given the retry options
+type RetryHandler struct {
```

Comment:
Since this is mostly pulled from [kiota](https://github.com/microsoft/kiota-http-go/blob/main/retry_handler.go), lets keep it in separate file.